### ISSUE TITLE
Email Subscription

### ISSUE THREAD

**Issue by ozkriff** [rust-gamedev/rust-gamedev.github.io#24]
> _@jeffvandyke:_
> I just came from [Newsletter #1](https://rust-gamedev.github.io/2019/09/02/newsletter-001.html) - very nice! I think some folks (including me) would appreciate a email subscription option like This Week In Rust does (twitter and reddit updates don't show up in my inbox). Perhaps whatever email service they use could be adopted for rust-gamedev?
>
> _@17cupsofcoffee:_
> [They apparently use MailChimp for this](https://github.com/cmr/this-week-in-rust#to-build-the-newsletter) - their free plan only allows up to 2000 subscribers, however, and [they already surpassed that in 2016](https://github.com/cmr/this-week-in-rust/issues/327), so they must be on one of the paid ones.
>
> I think this is definitely a good idea, but we'd need to either find a service with a more generous free plan and/or figure out a plan for how it'd be paid for.
>
> _@phaazon:_
> That makes me think I should really have an RSS stream on my website… I need to check how to do that.

(Extracted from https://github.com/rust-gamedev/rust-gamedev.github.io/issues/2)

**Comment by 17cupsofcoffee** [comment]
At the very least, we should make sure that we submit the newsletter for inclusion in This Week In Rust every time it comes out. [I think this was done for the first one](https://this-week-in-rust.org/blog/2019/09/03/this-week-in-rust-302/), not sure if that was a PR from one of us or someone on their end :)

**Comment by ozkriff** [comment]
About TWIR: It was [my PR](https://github.com/cmr/this-week-in-rust/pull/997), but they have a policy of not including newsletters and other regular issues. The first newsletter was included as a one-time exception.

**Comment by 17cupsofcoffee** [comment]
Ah, that's a bummer. I get it, though, they probably don't want their entire newsletter to be... links to other newsletters 😅

**Comment by JeanMertz** [comment]
We can at least ask to add it one more time, with a notice that we now have a proper newsletter to which people can subscribe (when that is in place, of course).

**Comment by 17cupsofcoffee** [comment]
So, as far as I can tell we've got a few options:

* MailChimp is pretty tried and tested. They offer up to 2000 subscribers, and then the next tier up is $9.99 a month for 50,000. That's not a huge expense, but we'd have to figure out where that money would come from. Good analytics features, etc.
    * There's several other companies at a similar scale to MailChimp (SendGrid, MailGun), but I haven't found any with more generous free tiers.
* TinyLetter is free, with a hard cap of 5000 subscribers. They're owned by MailChimp though, so you can just migrate if you hit the cap I think. Somewhat more limited feature set than other services, no analytics, etc. Might be a good option while we figure out what the demand is.
* buttondown.email has a more limited free tier of 1000 subscribers, with the next tier being $29 for (I think) unlimited. That's not great, but the main selling point here would be that buttondown emails are authored in Markdown, so we could literally just drop the post into the email with minimal hassle.

**Comment by AlexEne** [comment]
I'm fine to sponsor the 9.99 $ mail chimp payed tier.

**Comment by JeanMertz** [comment]
Although I'm not part of the WG, I'm following along and have a vested interest in it succeeding. We can split the bill if that helps, I can wire you 12 months upfront to make things less complicated.

**Comment by 17cupsofcoffee** [comment]
I'm also happy to chip in - just wanted to make sure I wasn't going to get stuck with the bill forever if I set it up 😛 That said, this is all theoretical until we actually hit 2k subscribers, which is a pretty high bar to cross!

I'll have a play with some of the above services over the weekend and see if I can figure out a reasonably ergonomic pipeline.

**Comment by 17cupsofcoffee** [comment]
One potential stumbling block I've run into - MailChimp [require you to provide a physical mailing address before you can set up a campaign](https://mailchimp.com/help/anti-spam-requirements-for-email/), and it has to be included in every email you send to comply with their TOS (and some countries' laws, I think?). I wouldn't be comfortable using my home address for that, I don't think I could get away with using my employer's, and setting up a PO box is pretty pricy.

**Comment by Lokathor** [comment]
Is there some sort of... mozilla office address thing we can give since we're part of the broader Rust team?

**Comment by ozkriff** [comment]
[Another email subscription request from Twitter](https://twitter.com/masonremaley/status/1180617286111023104):

@masonremaley:
> Is there a way to sign up to get this emailed to me whenever there’s a new one? (On mobile atm, apologies if there’s a button for this right in front of me and I’m just missing it!)

**Comment by ozkriff** [comment]
@AlexEne @erlend-sh @kvark hey, wg leads, do you know anyone we can ask about getting a group/official inbox for this thing?

**Comment by erlend-sh** [comment]
I’m not aware of any official service offer here. Amethyst org would be happy to sponsor such a thing though. It seems we’re at most talking about $10/month for something like Mailchimp?

**Comment by 17cupsofcoffee** [comment]
I did a writeup of the costs of different services here: https://github.com/rust-gamedev/rust-gamedev.github.io/issues/24#issuecomment-537960781

Last time we looked at this though, the main issue wasn't the cost of the emails, it was the requirement to provide a physical mailing address in every email (which is due to anti-spam legislation, I think). That'd require us to either:

* Use one of our personal/workplace addresses (which seems a bit dodgy)
* Pay for a PO box (which is pricy)
* Use some kind of Mozilla address (which seems less likely since the layoffs)
* Ignore the rules and hope they don't notice (which would be a bit embarrasing if we got shut down 😅)

**Comment by AlexEne** [comment]
Yes, the main blocker wasn't the subscription, but the requirement of a real and valid ;) mail address. This one has to be included in all newsletters at the bottom somewhere so using any of our personal ones isn't necessarely the best idea.

I'd propose we list some PO box prices in our countries and go for the cheapest one. I'm in the UK and Royal Mail is a bit pricey but there are some private PObox suppliers that go for a smaller subscription ( I need to investigate them again ).

**Comment by ozkriff** [comment]
Haven't used them so far, but PO boxes for individuals in my city (SPb) [seem to cost around 3$/mo (and 5$/mo for named ones)](https://abox.pochta.ru/cabinet/popup/DocTarifUI).

**Comment by zicklag** [comment]
In the interim we could suggest that people can subscribe to the forum topic where @ozkriff has been posting the newsletters:

https://users.rust-lang.org/t/this-month-in-rust-gamedev/45520?u=zicklag

**Comment by ozkriff** [comment]
Another [request from Twi](https://twitter.com/muhuk/status/1465664189226176513):

> I see the RSS feed, and I've added it to my reader. But these days I look at my feeds once or twice a year. Is there mailing list sort of thing?

**Comment by janhohenheim** [comment]
I added the `help wanted` label because I would love to have this feature, but do not have the expertise to implement it. If anyone is interested, hit me up :)

**Comment by ghenry** [comment]
Is there funding to pay for a service like Rust weekly use?

**Comment by janhohenheim** [comment]
Ping @AngelOnFira 
If not, I'm fine with spending some of my own money on this or adding a link for crowdfunding

**Comment by ghenry** [comment]
I would have thought that there would be budget for community services like this from the Rust Foundation?

They could fund the Rust Weekly one, then get the account holder to generate a new API key or rss to email feed/list and the html form snippet from Mailchimp for the Game dev website. 

I don't think anyone should tie this to their own account or funds.

**Comment by janhohenheim** [comment]
@ghenry that is probably the case, agreed. Just saying that money is not a blocker :)

**Comment by ghenry** [comment]
Then a one time config to enable:

https://mailchimp.com/features/rss-to-email/

**Comment by janhohenheim** [comment]
@ghenry that looks easier than I thought, thanks.
I can look into it once the April newsletter is done and someone tells me how the funding works. Maybe I'll shoot the TWIR people an email.

**Comment by ghenry** [comment]
Sounds good! I'm keenly watching :-)
